### DISCUSSION TITLE
Android 14 restore Database doesn't work

### DISCUSSION THREAD

**Discussion by rodrigomarquestnd** [NightscoutFoundation/xDrip#3149] [General]
The restore of a Database on android 14 don't work. At least for me.

**Comment by Navid200** [comment]
xDrip has two different database save and restore strategy.  Which one doesn't work for you?
What is the phone brand and model?
Which version of xDrip is on the phone?

**Reply by rodrigomarquestnd** [reply]
I have export the Database and copy from a Xiaomi Mi8 to a Pixel 8 Pro, then create a folder called xdrip and put the zipped file there. On xdrip try to restore and give the successful message but nothing happen, xdrip just close. I'm using the last version on both phones

**Reply by Navid200** [reply]
Try the alternative method:
https://navid200.github.io/xDrip/docs/GoogleDriveBackup.html

**Reply by rodrigomarquestnd** [reply]
It gives "restore succeeded" and then remains the same
![Screenshot_20231021-014256](https://github.com/NightscoutFoundation/xDrip/assets/96501526/f7d9fa23-0218-4d24-bcc8-a3e1722243aa)
![Screenshot_20231021-014411](https://github.com/NightscoutFoundation/xDrip/assets/96501526/14ebf198-1086-4b0e-88f7-1cb09e52ec06)

**Reply by Navid200** [reply]
What's the date of the last entry on the old phone?
What do you see in statistics on the new phone after restore?

**Reply by rodrigomarquestnd** [reply]
The xdrip is still working on the old phone. On the new phone the statistics doesn't appear 
![Screenshot_20231021-014821](https://github.com/NightscoutFoundation/xDrip/assets/96501526/f0083951-5f02-4de2-8864-464b5169936f)

**Reply by Navid200** [reply]
Please show the logs on the new phone.

**Reply by rodrigomarquestnd** [reply]
It could be this. My backup is over 1gb zipped, and Over 5gb unzipped
![Screenshot_20231021-015239](https://github.com/NightscoutFoundation/xDrip/assets/96501526/86bb208a-6b7a-4170-9754-e2193903a580)

**Reply by Navid200** [reply]
Why would you want to have so much data in xDrip?
When you go to see your doctor, you need the last 90 days.
Do you have Nightscout?
If you have Nightsocut, you can access all the way back.

But, why so much data in xDrip?

**Reply by rodrigomarquestnd** [reply]
I don't need this much data, since i use xdrip i just let it this way. How can i just use the last 90 days and erase the rest?

**Reply by Navid200** [reply]
When you buy a new phone, only transfer the settings, but, not the database.
So, every time you buy a new phone, you start from scratch.

As long as you upload to Nightsocut, you can always create your reports for your doctor from Nightscout.

I don't believe we have the option of deleting only a part of the database easily.

**Reply by rodrigomarquestnd** [reply]
Yeah, i think it's better that way, i'm going to do that. Thanks for your help!

**Reply by Navid200** [reply]
You're welcome
